packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -660,17 +660,38 @@ class _RailDestination extends StatelessWidget {
       selected: selected,
       child: Stack(
         children: <Widget>[
-          Material(
-            type: MaterialType.transparency,
-            child: InkResponse(
-              onTap: onTap,
+          // Material(
```

Comment:
We need to keep the Material and the InkResponse to draw proper reactions when this is interacted with.